kyo-prelude/shared/src/main/scala/kyo/kernel/Loop.scala

```diff
@@ -175,6 +401,18 @@ object Loop:
         loop(0)()
     end indexed
 
+    /** Executes an indexed loop with a single state value.
+      *
+      * Similar to the standard indexed loop, but also maintains one state value between iterations. Each iteration receives both the
+      * current index and state value, enabling more complex stateful computations with index tracking.
+      *
+      * @param input
+      *   The initial state value
+      * @param run
+      *   The function to execute for each iteration, receiving the current index and state, and producing an outcome
+      * @return
+      *   The final result after loop completion
+      */
     inline def indexed[A, O, S](input: A)(inline run: Safepoint ?=> (Int, A) => Outcome[A, O] < S)(using
```

Comment:
Do we need to specialize this function?

Reply:
It's not strictly necessary because a regular loop where the user updates the index manually should not box as well but it's more of a usability feature. Indexed loops are common.